### ISSUE TITLE
同一批测试数据，test-only 的accuracy和 训练时的validate accuracy 差很多？

### ISSUE THREAD

**Issue by peiji1981** [zhiqwang/image-captioning#9]
多gpu模式没有加入吗？

**Comment by zhiqwang** [comment]
test-only 计算 accuracy 是和 validate 一样的代码，准确率应该是一样的；我这边暂时没有计算资源，所以没有加入多 gpu 模式，对于多 gpu 模式，欢迎提pr。

**Comment by peiji1981** [comment]
感谢回复，代码一样我看到了，就是train期间的validate 有torch.no grad 有不同。还有就是 为什么validate/test 的数据需要分batch呢？ 还是说resume模型的时候有问题 造成accuracy有区别

**Comment by zhiqwang** [comment]
`--test-only` 应该是要加 `torch.no_grad()` 的，这是我代码的bug，我修改一下。validate/test 分 batch 主要是为了利用矩阵batch计算加速。

**Comment by peiji1981** [comment]
感谢，这边暂时没有问题了。后面我自己加个多gpu模式
